packages/fast-components-react-base/src/slider/slider.tsx

```diff
@@ -894,6 +908,99 @@ class Slider extends Foundation<SliderHandledProps, SliderUnhandledProps, Slider
         return dragValue;
     };
 
+    /**
+     * Handles touch dragging
+     */
+    private handleUpperThumbTouchStart = (e: React.TouchEvent): void => {
+        this.handleThumbTouchStart(e, SliderThumb.upperThumb);
+    };
+
+    private handleLowerThumbMouseTouchStart = (e: React.TouchEvent): void => {
+        this.handleThumbTouchStart(e, SliderThumb.lowerThumb);
+    };
+
+    private handleThumbTouchStart = (e: React.TouchEvent, thumb: SliderThumb): void => {
+        if (e.defaultPrevented || this.isBusyOrDisabled()) {
+            return;
+        }
+        
+        e.preventDefault();
+        (e.target as HTMLElement).focus();
+        window.addEventListener("touchend", this.handleTouchEnd);
+        window.addEventListener("touchcancel", this.handleTouchEnd);
+        window.addEventListener("touchmove", this.handleTouchMove);
+        this.setState({
+            isTouchDragging: true,
+            activeThumb: thumb,
+        });
+        const thisTouch: Touch = e.nativeEvent.touches.item(0);
+        this.updateDragValue(this.getTouchDragValue(thisTouch, thumb), thumb);
+    };
+
+    /**
+     *  Returns first valid touch found in a touch event
+     */
+    private getValidTouch = (event: TouchEvent): Touch => {
+        if (isNil(this.rootElement.current)) {
+            return null;
+        }
+        for (let i: number = 0; i < event.touches.length; i++) {
```

Comment:
It's best practice to always store the length or it will use the method to get it every loop iteration which is a slight perf problem, you can do it like this `for (let i: number = 0, touchLength: number = event.touches.length; i < touchLength; i++) `.

Additionally, if you aren't using the index for anything you could use [for...of](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of) instead, it reads a little better.

Reply:
yep.  Event.touches didn't work right off with the for...of syntax so didn't spend a lot of time on it as the iterator being a function call seemed odd (event.touches.item(index)).